### PR TITLE
MAINT: Upgrade the remaining GitHub Actions

### PR DESCRIPTION
**Pull request by cclauss**
### Description

- https://github.com/actions/checkout/releases
- https://github.com/docker/build-push-action/releases
- https://github.com/docker/login-action/releases
- https://github.com/docker/metadata-action/releases
- https://github.com/pypa/gh-action-pypi-publish/releases
- https://github.com/test-summary/action/releases

Also, add a final check to the CPython upgrade instructions.

### Checklists

- [ ] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- [ ] Add new / update outdated documentation

### CODE REVIEW

**Comment by cclauss** [comment]
Can someone please re-run just the one failing `test-core-firefox`?  That error looks flakey to me.

**Comment by hoodmane** [comment]
I reran the test. It indeed looks like a flake.

File: docs/development/maintainers.md [hunk 1]
```diff
@@ -125,3 +125,8 @@ For example: `v3.11.1` -> `v3.11.2`
    - `PYODIDE_IMAGE_TAG="20230301-chrome109-firefox109-py311"`
 8. Rebase any patches which do not apply cleanly.
 9. Create a pull request and fix any failing tests. This may be complicated for major releases of CPython.
+
+Once the process is complete, check the Python versions on both of these URLs:
+
+- https://pyodide.org/en/latest/console.html
+- https://pyodide.org/en/stable/console.html
```
Comment (hoodmane) [inline docs/development/maintainers.md:132]: This doesn't really seem very necessary to me.

**Review by hoodmane — APPROVED** [review]
Thanks for all your help @cclauss.

File: docs/development/maintainers.md (hunk 1 above)
Comment (cclauss) [reply docs/development/maintainers.md:132]: Try clicking them now...  They display counterintuitive versions of Python.  Stable should never be later than latest.

Comment (hoodmane) [reply docs/development/maintainers.md:132]: It's not though: stable is 3.10.2 and latest is 3.11.1

Comment (cclauss) [reply docs/development/maintainers.md:132]: Oh!  My misread!!

Comment (cclauss) [reply docs/development/maintainers.md]: Reverted.

**Comment by cclauss** [comment]
Again...
```
=================================== FAILURES ===================================
____________________ test_wrong_way_track_proxies[firefox] _____________________
src/tests/test_typeconversions.py:590: in test_wrong_way_track_proxies
    to_js(x, pyproxies=Object.new())
E   SystemError: error return without exception set
```
If this keeps repeating, perhaps we could modify:
* [src/tests/test_typeconversions.py#L589](https://github.com/pyodide/pyodide/blob/main/src/tests/test_typeconversions.py#L589)
```diff
-     with raises(TypeError):
+     with raises(SystemError, TypeError):
```

**Comment by hoodmane** [comment]
Well it's not acceptable for it to raise a `SystemError` there. But we can merge this with that error since it is unrelated to the changes here.

**Comment by hoodmane** [comment]
Thanks @cclauss!

**Pull request merged**